<launch>
	<!-- Command Line Arguments -->
	<arg name="cam_name" default="csi_cam" />                  <!-- The name of the camera (corrsponding to the camera info) -->
	<arg name="frame_id" default="/$(arg cam_name)_link" />    <!-- The TF frame ID. -->
	<arg name="sync_sink" default="true" />                    <!-- Synchronize the app sink. Setting this to false may resolve problems with sub-par framerates. -->
	<arg name="width" default="1280" />                        <!-- Image Width -->
	<arg name="height" default="720" />                       <!-- Image Height -->
	<arg name="fps" default="30" />                            <!-- Desired framerate. True framerate may not reach this if set too high. -->

	<node pkg="jetson_tensorrt" type="debug_node.py" name="rt_debug" output="screen"/>

	<!-- Make arguments available to parameter server -->
	<param name="$(arg cam_name)/image_width" type="int" value="$(arg width)" />
	<param name="$(arg cam_name)/image_height" type="int" value="$(arg height)" />
	<param name="$(arg cam_name)/target_fps" type="int" value="$(arg fps)" />


	<node pkg="nodelet" type="nodelet"
	    name="standalone_nodelet" args="manager"
	    output="screen">

			<!-- Define the GSCAM pipeline -->
			<env name="GSCAM_CONFIG" value="nvcamerasrc ! video/x-raw(memory:NVMM),
				width=(int)$(arg width), height=(int)$(arg height), format=(string)I420, framerate=(fraction)$(arg fps)/1 !
				nvvidconv flip-method=0 ! video/x-raw, format=(string)BGRx ! videoconvert ! video/x-raw, format=(string)BGR" />
	</node>

	<node pkg="nodelet" type="nodelet"
	    name="classifier"
	    args="load jetson_tensorrt/DIGITSClassifyNodelet standalone_nodelet"
	    output="screen">
			<param name="mean0" value="104.0069879317889"/>
			<param name="mean1" value="116.66876761696767"/>
			<param name="mean2" value="122.6789143406786"/>
	</node>

  <!-- Start the GSCAM nodelet -->
	<node pkg="nodelet" type="nodelet"
    name="$(arg cam_name)"
    args="load gscam/GSCamNodelet standalone_nodelet"
    output="screen">
		<param name="camera_name" value="$(arg cam_name)" />
    <param name="frame_id" value="$(arg frame_id)" />
    <param name="sync_sink" value="$(arg sync_sink)" />
    <remap from="camera/image_raw" to="$(arg cam_name)/image_raw" />
    <remap from="/set_camera_info" to="$(arg cam_name)/set_camera_info" />
  </node>

</launch>
